account_invoice_inter_company/tests/test_inter_company_invoice.py

```diff
@@ -342,6 +342,19 @@ def setUpClass(cls):
             cls.product_consultant_multi_company.company_ids = False
             cls.product_consultant_multi_company.product_tmpl_id.company_ids = False
 
+        cls.product_product_multi_company = cls.env["product.product"].create(
+            {
+                "name": "Product Multi Company",
+                "uom_id": cls.env.ref("uom.product_uom_unit").id,
+                "uom_po_id": cls.env.ref("uom.product_uom_unit").id,
+                "categ_id": cls.env.ref("product.product_category_2").id,
+                "type": "product",
+            }
+        )
```

Comment:
you could move the creation of the new product just after the creation of product_consultant_multi_company and then clear the field company_ids under the same 'if' condition in line 340